dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/workflow/step/FilesystemAccessingWorkflowStep.java

```diff
@@ -0,0 +1,94 @@
+package edu.harvard.iq.dataverse.workflow.step;
+
+import edu.harvard.iq.dataverse.workflow.WorkflowContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.UUID.randomUUID;
+
+/**
+ * Base class for {@link WorkflowStep}'s operating on a local filesystem.
+ * <p>
+ * Handles creation of a local working directory to be used for all file storage by the step.
+ * Also takes care for passing through that working directory to the next step of the same type.
+ * This way you can easily have multiple steps working in the same directory one after another,
+ * as long as each of them extend this class. To override this simply set {@value WORK_DIR_PARAM_NAME}
+ * input parameter to a different path you need.
+ */
+public abstract class FilesystemAccessingWorkflowStep implements WorkflowStep {
+
+    /**
+     * Directory to run execute the step in.
+     * If not defined a temporary directory will be created for that purpose.
+     */
+    public static final String WORK_DIR_PARAM_NAME = "workDir";
+
+    protected static final Logger log = LoggerFactory.getLogger(FilesystemAccessingWorkflowStep.class);
+
+    private final String workDirParam;
+
+    public FilesystemAccessingWorkflowStep(Map<String, String> inputParams) {
+        this.workDirParam = inputParams.get(WORK_DIR_PARAM_NAME);
+    }
+
+    @Override
+    public final WorkflowStepResult run(WorkflowContext context) {
+        try {
+            Path workDir = createWorkDir(context);
+
+            WorkflowStepResult.Source resultSupplier = runInternal(context, workDir);
+
+            return resultSupplier.apply(defaultOutputParams(workDir));
+        } catch (Exception e) {
+            return handleError(e);
+        }
+    }
+
+    protected abstract WorkflowStepResult.Source runInternal(WorkflowContext context, Path workDir) throws Exception;
+
+    protected WorkflowStepResult handleError(Exception e) {
+        log.error("Failed workflow step", e);
+        return new Failure(e.getMessage());
+    }
+
+    @Override
+    public WorkflowStepResult resume(WorkflowContext context, Map<String, String> internalData, String externalData) {
+        return new Success();
```

Comment:
I'm thinking if we should support that kind of operation. This will silently mark this step as it was executed normally, but step is not really resumable - maybe some kind of exception like `UnsupportedOperationException` will be more appropriate here? (maybe I'm missing a big picture here)